src/olympia/abuse/cinder.py

```diff
@@ -456,13 +456,29 @@ def flag_for_human_review(self, appeal=False):
             if appeal
             else NeedsHumanReview.REASON_ABUSE_ADDON_VIOLATION
         )
+        nhr_object = NeedsHumanReview(
+            version=self.version, reason=reason, is_active=True
+        )
         if self.version:
-            NeedsHumanReview.objects.get_or_create(
+            if not NeedsHumanReview.objects.filter(
                 version=self.version, reason=reason, is_active=True
-            )
+            ).exists():
+                nhr_object.save(_no_automatic_activity_log=True)
+                versions_to_log = [self.version]
+            else:
+                versions_to_log = []
         else:
-            self.addon.set_needs_human_review_on_latest_versions(
-                reason=reason, ignore_reviewed=False, unique_reason=True
+            versions_to_log = self.addon.set_needs_human_review_on_latest_versions(
```

Comment:
What is the point of skipping activity log here? If the activity log is implicit in the method, why not let it be handeld there?

The if versions_to_log could be handled in the existing version branch above? Seems like a smaller change.

Reply:
I see. I guess I'm wondering why all of the acrobatics with skipping automatic activity? If the problem is that we want sometimes to save with an automatic activity and sometimes with another activity, and we already have that logic implemented on the save method, then why not make that configurable in the save method? Just seems like I had to do a lot of reading around what was going on to figure it out. If you modified the save method to accept a type of activity log, then you could just 

```python
nhr_object.save(_activity_log=amo.LOG.NEEDS_HUMAN_REVIEW_CINDER)
```

And that could be encapsulated in the _set_blah_blah method on the model, skipping the passing around this flag and then doing the same thing back up in this class.

Reply:
We'd lose the second advantage of a single activity log for multiple versions (the reviewer tools is the other place that passes `_no_automatic_activity_log`, for that same reason, to be able to save a single log and link it to multiple versions)